sklearn/linear_model/_ridge.py

```diff
@@ -1459,43 +1473,40 @@ def fit(self, X, y, sample_weight=None):
         else:
             sqrt_sw = np.ones(X.shape[0], dtype=X.dtype)
 
+        X_mean, *decomposition = decompose(X, y, sqrt_sw)
+
         scorer = check_scoring(self, scoring=self.scoring, allow_none=True)
         error = scorer is None
 
         n_y = 1 if len(y.shape) == 1 else y.shape[1]
-        cv_values = np.zeros((n_samples * n_y, len(self.alphas)),
-                             dtype=X.dtype)
-        C = []
-        X_mean, *decomposition = decompose(X, y, sqrt_sw)
+
+        if self.store_cv_values:
+            self.cv_values_ = np.empty(
+                (n_samples * n_y, len(self.alphas)), dtype=X.dtype)
+
+        best_coef, best_score, best_alpha = None, None, None
+
         for i, alpha in enumerate(self.alphas):
             G_inverse_diag, c = solve(
                 float(alpha), y, sqrt_sw, X_mean, *decomposition)
             if error:
                 squared_errors = (c / G_inverse_diag) ** 2
-                cv_values[:, i] = squared_errors.ravel()
+                alpha_score = -squared_errors.mean()
```

Comment:
I think there's another bug: we need to devide sample_weight here, otherwise we get weighted error.

Reply:
Why shouldn't sample weights be taken into account when computing the error? Note that this has always been the behaviour of the GCV estimator and is made explicit here

https://github.com/scikit-learn/scikit-learn/blob/98cb91b545f9d630b2d37de996cccbe35a33695b/sklearn/linear_model/_ridge.py#L1554

Reply:
It's embarassing that such important thing is documented in private functions.
I think users will expect that ``RidgeCV()`` should be equivalent to ``GridSearchCV(Ridge(), cv=LeaveOneOut())``